logstash-core/lib/logstash/pipeline.rb

```diff
@@ -657,18 +610,25 @@ def shutdown_workers
   def filter(event, &block)
     maybe_setup_out_plugins
     # filter_func returns all filtered events, including cancelled ones
-    filter_func(event).each {|e| block.call(e)}
+    wait_until_started
```

Comment:
This function (filter) seems to have a downstream code creation impact with several `debug` checks (TerminalDebugDataset, CompiledPipeline.buildFilterFuncDebug, etc).

Is it necessary? If it exists only to satisfy parts of the existing test suite, then we're testing a kind of test fixture and skipping testing parts of the actual implementation. Can you help me understand better?